apps/web/pages/api/addresses/[address]/usernamesDiscountMessage/index.ts

```diff
@@ -0,0 +1,112 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { kv } from '@vercel/kv';
+import { hasRegisteredWithDiscount } from 'apps/web/src/components/Usernames/registrarController';
+import { ethers } from 'ethers';
+import {
+  trustedSignerAddress,
+  trustedSignerPKey,
+} from 'apps/web/src/components/Usernames/constants';
+
+type PreviousClaim = {
+  address: string;
+  signedMessage: string;
+};
+
+type LinkedAddresses = {
+  idemKey: string;
+  linkedAddresses: string[];
+};
+
+const expiry = 300; //  5 minutes in seconds
+const previousClaimsKVPrefix = 'username:claims:';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== 'GET') {
+    res.status(405).json({ error: 'method not allowed' });
+    return;
+  }
+  const { address } = req.query;
+  if (!address) {
+    return res.status(400).json({ error: 'address is required' });
+  }
+
+  let linkedAddressResponse: LinkedAddresses;
+  try {
+    //hit CDP to get linked addresses
+    linkedAddressResponse = await getLinkedAddresses(address as string);
+
+    // check onchain if any linked address has registered previously
+    const hasPreviouslyRegistered = await hasRegisteredWithDiscount(
+      linkedAddressResponse.linkedAddresses,
+    );
+    // if any linked address registered previously return an error
+    if (hasPreviouslyRegistered) {
+      return res.status(409).json({ result: 'user has already claimed a username' });
```

Comment:
have we discussed with folks building out the FE whether its preferred to have an error vs some empty state response?